tools/rostest/scripts/rostest

```diff
@@ -33,4 +33,30 @@
 
 from rostest import rostestmain
 
-rostestmain()
+import os
+import sys
+import tempfile
+
+
+class PythonPathOverride:
+    """Replace python so test scripts with unmodified shebangs use correct python version."""
+
+    def __enter__(self):
+        # skip this hack on windows
+        if 'nt' == os.name:
+            return
+
+        self.dir = tempfile.TemporaryDirectory(prefix="rostest_bin_hook")
+        os.symlink(os.path.abspath(sys.executable), os.path.join(self.dir.name, "python"))
```

Comment:
Nitpick: [sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) is already an absolute path.

Reply:
resolved in 6e57fa0